### PR TITLE
htlcswitch: move hop iterator into htlcswitch/hop package

### PR DESCRIPTION
**Pull request by joostjager**
Prepares for onion blob decoding outside of htlcswitch.

### CODE REVIEW

File: htlcswitch/failure.go [hunk 1]
```diff
@@ -47,203 +46,15 @@ type ErrorDecrypter interface {
...
-
-	return nil
-
-}
-
-// A compile time check to ensure SphinxErrorEncrypter implements the
-// ErrorEncrypter interface.
-var _ ErrorEncrypter = (*SphinxErrorEncrypter)(nil)
-
 // OnionErrorDecrypter is the interface that provides onion level error
```
Comment (cfromknecht) [inline htlcswitch/failure.go]: not critical to this pr, but is this still the right place for `OnionErrorDecrypter`?

File: htlcswitch/hop/iterator.go [hunk 2]
```diff
@@ -80,31 +79,31 @@ func (r *sphinxHopIterator) EncodeNextHop(w io.Writer) error {
 //
 // NOTE: Part of the HopIterator interface.
 func (r *sphinxHopIterator) ForwardingInstructions() (
-	hop.ForwardingInfo, error) {
+	ForwardingInfo, error) {
```
Comment (cfromknecht) [inline htlcswitch/hop/iterator.go]: nit: 78 chars

File: htlcswitch/link.go [hunk 3]
```diff
@@ -2553,7 +2553,7 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 	l.tracef("processing %d remote adds for height %d",
 		len(lockedInHtlcs), fwdPkg.Height)
 
-	decodeReqs := make([]DecodeHopIteratorRequest, 0, len(lockedInHtlcs))
+	decodeReqs := make([]hop.DecodeHopIteratorRequest, 0, len(lockedInHtlcs))
```
Comment (cfromknecht) [inline htlcswitch/link.go]: nit: long line?

File: htlcswitch/failure.go (hunk 1 above)
Comment (joostjager) [reply htlcswitch/failure.go]: yeah, the same thought crossed my mind, but I wanted to do the minimum required for the receiver side.

**Pull request merged**